### ISSUE TITLE
Support React 18.0

### ISSUE THREAD

**Issue by nikivdev** [alibaba/hooks#1532]
After updating react package to 18.0 and running `npm install` get this:

```
npm ERR! Found: react@18.0.0
npm ERR! node_modules/react
npm ERR!   react@"18.0.0" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^16.8.6 || ^17.0" from ahooks@3.2.0
npm ERR! node_modules/ahooks
npm ERR!   ahooks@"^3.2.0" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

**Comment by brickspert** [comment]
Released in v3.3.0! try it~

**Comment by nikivdev** [comment]
It works, thanks.
